Ryan Implemented AttributeStrategy Entity and Related Testing Files

not-ryan-ning opened the pull request:
Finished implementation of the AttributeStrategy entity and related JUnit tests. 

- Implemented the AttributeStrategy entity by calculating the Euclidean distances between the attributes in each playlist as follows: 

1. EuclideanDistanceAttribute = (playlist1.getAttribute - playist2.getAttribute) ^ 2
2. Sum all the Euclidean distances, and take the square root of the sum for the total distance between all attributes. 

- The JUnit tests test a 100%, 50%, and 0% similarity between the 2 Playlist Objects. It also tests that the AttributeStrategy object itself can be instantiated without problems. The tests in question all pass. 
![Screenshot 2023-12-02 at 6 45 27 PM](https://github.com/not-ryan-ning/Friendify/assets/136235474/a1660669-f52d-440c-a509-0323373dd7b1)

zfjuliana reviewed (commented):
Hi Ryan, your code, comments, and tests style are organized and easy to follow. For the AttributeStrategy, I see you chose to use the Euclidean distance algorithm to calculate the similarity score. However, do you think it would be possible to use an algorithm that returns a similarity score between 0-1, with a score of 1 being the most similar, or normalize the result? This way, the weighted similarity score across all strategies can be determined (code is located in the MatchInteractor class). Thank you!

not-ryan-ning commented:
Changed the return of similarity score so it is normalized to range between 0.0 and 1.0, and is 1.0 when the playlists are the most similar and 0.0 when the playlists are most dissimilar in terms of attributes. Fixed tests accordingly to expect those values, and renamed one test case for clarity. Tests are passing! Please let me know if there are any other problems @zfjuliana 

![Screenshot 2023-12-02 at 9 30 21 PM](https://github.com/not-ryan-ning/Friendify/assets/136235474/1f0a361b-f50a-46ee-ad58-c7316afbbfb6)

zfjuliana commented:
Looks good! I will approve the merge
